src/ui/ui_review_menu_nbgl.c

```diff
@@ -0,0 +1,335 @@
+            txInfos.fields[1].value = fromAddress;
+            pairList.nbPairs = 2;
+            txInfos.flowTitle = "Review";
+            txInfos.flowSubtitle = "Permission Update";
+            infoLongPress.text = "Confirm Update";
+            break;
+        case APPROVAL_EXCHANGE_CREATE:
+            txInfos.fields[0].item = "Token 1";
+            txInfos.fields[0].value = fullContract;
+            txInfos.fields[1].item = "Amount 1";
+            txInfos.fields[1].value = (char *) G_io_apdu_buffer;
+            txInfos.fields[2].item = "Token 2";
+            txInfos.fields[2].value = toAddress;
+            txInfos.fields[3].item = "Amount 2";
+            txInfos.fields[3].value = (char *) G_io_apdu_buffer + 100;
+            txInfos.fields[4].item = "From Address";
+            txInfos.fields[4].value = fromAddress;
+            pairList.nbPairs = 5;
+            txInfos.flowTitle = "Review Exchange";
+            infoLongPress.text = "Accept and Create\nExchange";
+            break;
+        case APPROVAL_EXCHANGE_TRANSACTION:
+            txInfos.fields[0].item = "Exchange ID";
+            txInfos.fields[0].value = toAddress;
+            txInfos.fields[1].item = "Token pair";
+            txInfos.fields[1].value = fullContract;
+            txInfos.fields[2].item = "Amount";
+            txInfos.fields[2].value = (char *) G_io_apdu_buffer;
+            txInfos.fields[3].item = "Expected";
+            txInfos.fields[3].value = (char *) G_io_apdu_buffer + 100;
+            txInfos.fields[4].item = "From Address";
+            txInfos.fields[4].value = fromAddress;
+            pairList.nbPairs = 5;
+            break;
+        case APPROVAL_EXCHANGE_WITHDRAW_INJECT:
+            txInfos.fields[0].item = "Action";
+            txInfos.fields[0].value = (char *) G_io_apdu_buffer + 100;
+            txInfos.fields[1].item = "Exchange ID";
+            txInfos.fields[1].value = toAddress;
+            txInfos.fields[2].item = "Token Name";
+            txInfos.fields[2].value = fullContract;
+            txInfos.fields[3].item = "Amount";
+            txInfos.fields[3].value = (char *) G_io_apdu_buffer;
+            txInfos.fields[4].item = "From Address";
+            txInfos.fields[4].value = fromAddress;
+            pairList.nbPairs = 5;
+            break;
+        case APPROVAL_WITNESSVOTE_TRANSACTION:
+            uint8_t i;
+            if(votes_count > MAX_TX_FIELDS - 2) {
+                THROW(E_INCORRECT_DATA);
+            }
+            for (i = 0; i < votes_count; i++) {
+                txInfos.fields[i].item = (char *) (G_io_apdu_buffer + voteSlot(i, VOTE_ADDRESS));
+                txInfos.fields[i].value = (char *) (G_io_apdu_buffer + voteSlot(i, VOTE_AMOUNT));
+            }
+            txInfos.fields[i].item = "Total Vote Count";
+            txInfos.fields[i].value = fullContract;
+            txInfos.fields[i + 1].item = "From Address";
+            txInfos.fields[i + 1].value = fromAddress;
```

Comment:
```suggestion
            txInfos.fields[votes_count].item = "Total Vote Count";
            txInfos.fields[votes_count].value = fullContract;
            txInfos.fields[votes_count + 1].item = "From Address";
            txInfos.fields[votes_count + 1].value = fromAddress;
```
I think that it makes the added check on fields array lenght easier to verify?
But up to you.

Reply:
I don't understand